inc/classes/optimization/class-abstract-optimization.php

```diff
@@ -117,4 +117,20 @@ protected function get_file_path( $url ) {
 	protected function get_file_content( $file ) {
 		return rocket_direct_filesystem()->get_contents( $file );
 	}
+
+	/**
+	 * Hides unwanted blocks from the HTML to be parsed for optimization
+	 *
+	 * @since 3.1.4
+	 * @author Remy Perona
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	protected function hide_comments( $html ) {
+		$html = preg_replace( '#<!--\s?noptimize\s?-->.*?<!--\s?/\s?noptimize\s?-->#is', '', $html );
```

Comment:
`\s?`: are you sure to have only 1 white space each time? I personally would go for `\s*` to be safe.